extensions/vscode-api-tests/src/singlefolder-tests/terminal.test.ts

```diff
@@ -96,124 +98,127 @@ import { assertNoRpc } from '../utils';
 			});
 		});
 
-		test('onDidCloseTerminal event fires when terminal is disposed', (done) => {
-			disposables.push(window.onDidOpenTerminal(term => {
-				try {
-					equal(terminal, term);
-				} catch (e) {
-					done(e);
-					return;
-				}
-				terminal.dispose();
-				disposables.push(window.onDidCloseTerminal(() => done()));
-			}));
+		test('onDidCloseTerminal event fires when terminal is disposed', async () => {
 			const terminal = window.createTerminal();
+			const result = await new Promise<Terminal>(r => {
+				disposables.push(window.onDidOpenTerminal(r));
+			});
+			equal(result, terminal);
+			await new Promise<void>(r => {
+				disposables.push(window.onDidCloseTerminal(t => {
+					if (t === terminal) {
+						r();
+					}
+				}));
+				terminal.dispose();
+			});
 		});
 
-		test('processId immediately after createTerminal should fetch the pid', (done) => {
-			disposables.push(window.onDidOpenTerminal(term => {
-				try {
-					equal(terminal, term);
-				} catch (e) {
-					done(e);
-					return;
-				}
-				terminal.processId.then(id => {
-					try {
-						ok(id && id > 0);
-					} catch (e) {
-						done(e);
-						return;
-					}
-					terminal.dispose();
-					disposables.push(window.onDidCloseTerminal(() => done()));
-				});
-			}));
+		test('processId immediately after createTerminal should fetch the pid', async () => {
 			const terminal = window.createTerminal();
-		});
+			const result = await new Promise<Terminal>(r => {
+				disposables.push(window.onDidOpenTerminal(t => {
+					if (t === terminal) {
+						r(t);
+						t.processId.then(id => id && id > 0);
```

Comment:
We can await this at the top level, right now `id && id > 0` failing wouldn't throw as well since they no longer use `ok`